### DISCUSSION TITLE
Include @livewirescripts in <head> with defer attribute

### DISCUSSION THREAD

**Discussion by SimonBroekaert** [livewire/livewire#3001] [7. Bugs (archived)]
**Problem/Feature request**

I don't find any way to include the livewire scripts in the head of my HTML with the defer attribute.
All page speed tests show that putting your scripts in the head with defer speeds up your page alot.

So such a feature would be very much appreciated! Keep up to great work! 🥇 

![image](https://user-images.githubusercontent.com/35606498/120510734-ff0fcb00-c3c9-11eb-886d-e667ff059d0b.png)

**Comment by Rhincodon** [comment]
I have a project where livewire affects page performance score before of the current way of loading:

![image](https://user-images.githubusercontent.com/6630959/125779732-f0191981-e44b-442d-95a6-f535da73f07c.png)

And because pagespeed is important for SEO (Google rolls out some updates for this) I think it's important to change the load behaviour for Livewire as well.

**Comment by joshhanley** [comment]
@SimonBroekaert and @Rhincodon I have done that with my blog. It is a work around, but I will outline how I achieved it below.

Put this blade directive in your `AppServiceProvider`

```php
Blade::directive('livewireScriptPath', function () {
    $appUrl = config('livewire.asset_url', rtrim($options['asset_url'] ?? '', '/'));

    $manifest = json_decode(file_get_contents(base_path('vendor/livewire/livewire/dist/manifest.json')), true);
    $versionedFileName = $manifest['/livewire.js'];

    $fullAssetPath = "{$appUrl}/livewire{$versionedFileName}";

    if (file_exists(public_path('vendor/livewire/manifest.json'))) {
        $publishedManifest = json_decode(file_get_contents(public_path('vendor/livewire/manifest.json')), true);
        $versionedFileName = $publishedManifest['/livewire.js'];

        $fullAssetPath = ((($_ENV['SERVER_SOFTWARE'] ?? null) === 'vapor') ? config('app.asset_url') : $appUrl).'/vendor/livewire'.$versionedFileName;
    }

    return $fullAssetPath;
});
```

Then in your layout file, put this in your head tag
```html
<head>
    <!-- other head stuff here -->
    <link rel="preload" as="script" href="@livewireScriptPath">
</head>
```

And leave the `@livewireScripts` or `<livewire:scripts />` at the end of your body tag.

Hope this helps!

**Comment by ThisGitHubUsernameWasAvailable** [comment]
Yes! I was very surprised to see the livewire scripts inserted before the closing `<body>`. It's a very bad practice. Scripts should be placed in `<head>` with a `defer` attribute, so performance will be much better.
